code/obj/item.dm

```diff
@@ -475,14 +475,24 @@
 /obj/item/proc/equipment_click(atom/source, atom/target, params, location, control, origParams, slot) //Called through hand_range_attack on items the mob is wearing that have HAS_EQUIP_CLICK in flags.
 	return 0
 
-/obj/item/proc/combust() // cogwerks- flammable items project
+/obj/item/proc/combust(obj/item/W as obj) // cogwerks- flammable items project
 	if(processing_items.Find(src)) //processing items cant be lit on fire to avoid weird bugs
 		return
 	if (!src.burning)
 		START_TRACKING_CAT(TR_CAT_BURNING_ITEMS)
 		src.visible_message("<span class='alert'>[src] catches on fire!</span>")
 		src.burning = 1
 		src.firesource = TRUE
+		if (istype(src, /obj/item/plant))
+			if (!ON_COOLDOWN(global, "hotbox_adminlog", 30 SECONDS)) //blocks every instance of the plant logging
+				var/list/hotbox_plants = list()
+				for (var/obj/item/plant/P in get_turf(src))
+					hotbox_plants += P
+					if (length(hotbox_plants) == 5) //number is up for debate, 5 seemed like a good starting place
+						if (W && W.firesource)
+							message_admins("([src]) was set on fire on the same turf as at least ([length(hotbox_plants)]) other plants at [log_loc(src)] by item ([W]). Last touched by: [W.fingerprintslast ? "[W.fingerprintslast]" : "*null*"].")
+						else
+							message_admins("([src]) was set on fire on the same turf as at least ([length(hotbox_plants)]) other plants at [log_loc(src)].")
```

Comment:
looking pretty good. If we bring the if in some we can get a full count, also instead of checking if `W` exists and then reading it's `firesource` var we can use the `?.` operator which is identical in behavior in the case of `(W && W.firesource)` while also slightly faster
```suggestion
				if (length(hotbox_plants) >= 5) //number is up for debate, 5 seemed like a good starting place
					if (W?.firesource)
						message_admins("([src]) was set on fire on the same turf as at least ([length(hotbox_plants)]) other plants at [log_loc(src)] by item ([W]). Last touched by: [W.fingerprintslast ? "[W.fingerprintslast]" : "*null*"].")
					else
						message_admins("([src]) was set on fire on the same turf as at least ([length(hotbox_plants)]) other plants at [log_loc(src)].")
```